shrinkwrap/elf.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import os
+import re
+from abc import ABC, abstractmethod
+from typing import Dict, Iterable, Optional
+
+import lief  # type: ignore
+from sh import Command  # type: ignore
+
+from shrinkwrap import ldsoconf
+
+
+class LinkStrategy(ABC):
+    @staticmethod
+    def select_by_name(name: str) -> LinkStrategy:
+        if name == "native":
+            return NativeLinkStrategy()
+        elif name == "virtual":
+            return VirtualLinkStrategy()
+        else:
+            raise Exception(f"Unknown strategy: {name}")
+
+    @abstractmethod
+    def explore(self, binary: lief.Binary, filename: str) -> Dict[str, str]:
+        """
+        Determine the linking for all needed objects
+        """
+        pass
+
+
+class NativeLinkStrategy(LinkStrategy):
+    """Uses the native interpreter in the binary to determine the linking"""
+
+    def explore(self, binary: lief.Binary, filename: str) -> Dict[str, str]:
+        interpreter = Command(binary.interpreter)
+        resolution = interpreter("--list", filename)
+        result = {}
+        # TODO: Figure out why `--list` and `ldd` produce different outcomes
+        # specifically for the interpreter.
+        # https://gist.github.com/fzakaria/3dc42a039401598d8e0fdbc57f5e7eae
+        for line in resolution:
```

Comment:
@trws let me know if you have some knowledge here of this.

Reply:
That's a new one on me.  Maybe the loader just prints itself instead of looking it up?